src/containers/Application/ApplicationPageWrapper.tsx

```diff
@@ -111,4 +164,83 @@ function processRedirect(appState: any): void {
   }
 }
 
+interface buildProgressInApplicationProps {
+  elementsState: ApplicationElementStates | undefined
+  responses: ResponsesFullByCode | undefined
+  templateSections: TemplateSectionPayload[]
+  isLinear: boolean
+  currentSection: number
+  currentPage: number
+}
+
+function buildProgressInApplication({
+  elementsState,
+  responses,
+  templateSections,
+  isLinear,
+  currentSection,
+  currentPage,
+}: buildProgressInApplicationProps): ProgressInApplication {
+  if (!elementsState || !responses) return []
+  let sectionsStructure: ProgressInApplication = templateSections.map((section) => {
+    // Create an array with all pages in each section
+    const pages = Array.from(Array(section.totalPages).keys(), (n) => n + 1)
+    const checkIsCurrentPage = (page: number) =>
+      section.index === currentSection && page === currentPage
+    const checkIsPreviousPage = (index: number, page: number) =>
+      index < currentSection ? true : index === currentSection && page <= currentPage
```

Comment:
i see you quite like conditional operators. typically we would just add brackets around `index === currentSection && page <= currentPage` with an or in between.

I think what we want for canNavigate logic, for linear (does not need to be aware of current or previous page):

the first not valid or incomplete page from the start = set flag for cant navigate, then every page and section after that in linear mode, can't be navigated to

Reply:
Yes, the conditional helps me to think, but I agree is not the easiest to read. I'll change to what you suggest.
I will also consider what you comment on the status for `canNavigate` flag.

Reply:
> the first not valid or incomplete page from the start = set flag for cant navigate

I think this would be more complicated since we can have any page with status INVALID, that's why I'm checking first for `isPreviousPage`. Although if we want to display all validation in the progress bar after the user has been to the `Summary section` I don't think this code would work. But I'm not sure how we can change it to consider this overall validation.

Reply:
Doesn't `canNavigate` only apply to linear mode? So in linear mode you can only navigate to the first page that is invalid/incomplete ?